api/v1/openapi.yaml

```diff
@@ -831,7 +831,20 @@ paths:
             "$ref": "#/definitions/BPFMap"
         '404':
           description: Map not found
-
+  "/map/{name}/events":
```

Comment:
Are there any limiting parameters we might want to add? A "since" field? Count?

Even cooler would be streaming, of course...

Reply:
>Even cooler would be streaming, of course... 

Yeah this is a good idea, I was thinking about it initially but decided to go for a simpler approach. But now I'm thinking it may solve some potential issues I've been concerned about such as:

* Marshalling json requires basically a full copy of the buffers elements, we've run into OOM issues during sysdumps before. With streaming we could write these to the socket message by message.
* Updates and events must be 100% serialized, so they're all using the same mutex lock in bpf.Map. This is not a problem when adding events but could become an issue when configuring large buffers (i.e. we need to retain like 24h+ worth of events) which would result in blocking all map operations while the buffer is being dumped.

Reply:
Update: Reworked this to include streaming, the API no longer serves a single JSON document but rather json delimited "Event" models. This solves the problem stated above since events can be directly encoded/written one-by-one.